src/mmap.rs

```diff
@@ -410,6 +410,22 @@ impl GuestMemoryMmap {
     ///               The regions shouldn't overlap and they should be sorted
     ///               by the starting address.
     pub fn from_regions(mut regions: Vec<GuestRegionMmap>) -> result::Result<Self, Error> {
+        Self::from_arc_regions(regions.drain(..).map(Arc::new).collect())
+    }
+
+    /// Creates a new `GuestMemoryMmap` from a vector of Arc regions.
+    ///
+    /// Similar to the constructor from_regions() as it returns a
+    /// GuestMemoryMmap. The need for this constructor is to provide a way for
+    /// consumer of this API to create a new GuestMemoryMmap based on existing
+    /// regions coming from an existing GuestMemoryMmap instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `regions` - The vector of Arc regions.
+    ///               The regions shouldn't overlap and they should be sorted
+    ///               by the starting address.
+    pub fn from_arc_regions(regions: Vec<Arc<GuestRegionMmap>>) -> result::Result<Self, Error> {
```

Comment:
Just one question, should this be `from_regions` or `from_arc_regions`? I am not familiar enough with Rust API design to know the answer.

Reply:
What do you mean? I introduced a new constructor and since I was trying to keep backward compatibility, I didn't change the name of the original function `from_regions()`.

Reply:
Too much C++, Rust doesn't support function overloading. :)